clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -154,6 +155,28 @@ public class AzureComputeClient extends AzureBaseClient {
     null
   }
 
+  ServiceResponse<Void> destroyServerGroup(String resourceGroupName, String serverGroupName) {
+    ServiceResponse<Void> result = null
+    // The API call might return a timout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
+      try {
+        operationRetry ++
+        result = this.client.getVirtualMachineScaleSetsOperations()?.delete(resourceGroupName, serverGroupName)
+        operationRetry = AZURE_ATOMICOPERATION_RETRY
+      } catch (Exception e) {
```

Comment:
What if the exception is a cloud exception and the status code is a 404? Would you want to keep trying to delete it if it's already gone?